etc/esp-idf/README.md

```diff
@@ -0,0 +1,7 @@
+# ESP-IDF Build Instructions
+The ESP-IDF has its own CMake based build environment for application developers. Using the standard ESP-IDF application build environment, The openmrn/etc/esp-idf directory can be linked into the ESP application "components" directory.
+```
+cd <application path>/components
+ln -s <openmrn path>/etc/esp-idf/ openmrn
+```
+This will result in openmrn being compiled as a registered idf component with the name "openmrn".
```

Comment:
We should also document using idf_component.yml which is the approach recommended by Espressif for dependency management. 

A version of this could be:
```
dependencies:
  OpenMRN:
    version: {branch, hash, tag, etc}
    git: https://github.com/bakerstu/openmrn.git
    path: etc/esp-idf
```
The `path` line overrides where `idf.py` looks for the CMakeLists.txt within the referenced repository (it defaults to root of the repository).

If you include `override_path` it will skip looking at the repository and use a local copy instead, which can use environment variables, such as: `${OPENMRNPATH}/etc/esp-idf`.

The  `version` line is optional, when not present `idf.py build` will collect the latest git hash and record it in `dependencies.lock`. If there is a newer version available in git it will present the following during build:
```
-- Building ESP-IDF components for target esp32s3
NOTICE: 
Following dependencies have new versions available:
Dependency "OpenMRNIDF": "5d7b0b3157e3411b3f71ccfacedaef6cb1202bff" -> "63e2b2e3f6ba34719f50194240113247793ac3ac"
Consider running "idf.py update-dependencies" to update your lock file.
```